diff --git a/dist/index.mjs b/dist/index.mjs
index b4b69789af2d22e7761360b493b9e6f9abd03cfe..8685e4c10f1a55b9860113aca692825a8ddcf733 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -64,7 +64,7 @@ function createPreloadPlugin(options) {
                 const length = code.length;
                 if (it.type === "declaration") {
                   code = code.slice(0, it.start + offset) + it.code.replace(/export\s*/, "") + code.slice(it.end + offset);
-                  names.push(...it.names);
+                  names.push(it.name);
                 } else if (it.type === "named") {
                   code = code.slice(0, it.start + offset) + code.slice(it.end + offset);
                   names.push(...it.names);
